pkg/deploy/migration/on-reconcile-one-time-migration.go

```diff
@@ -0,0 +1,287 @@
+// As the function doesn't know the kind of the object with given name an empty object should be passed,
+// for example: addRequiredLabelsForObject(ctx, "my-secret", &corev1.Secret{})
+func addRequiredLabelsForObject(ctx *deploy.DeployContext, objectName string, obj client.Object) error {
+	// Check if the object is already migrated
+	if exists, _ := deploy.GetNamespacedObject(ctx, objectName, obj); exists {
+		// Default client sees the object in cache, no need in adding labels
+		return nil
+	}
+
+	err := ctx.ClusterAPI.NonCachingClient.Get(context.TODO(), types.NamespacedName{Namespace: ctx.CheCluster.Namespace, Name: objectName}, obj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// The object doesn't exist in cluster, nothing to do
+			return nil
+		}
+		return err
+	}
+
+	// Add required labels to the object
+	labels := obj.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	for labelName, labelValue := range deploy.GetLabels(ctx.CheCluster, deploy.DefaultCheFlavor(ctx.CheCluster)) {
+		labels[labelName] = labelValue
+	}
+	obj.SetLabels(labels)
+	if err := ctx.ClusterAPI.NonCachingClient.Update(context.TODO(), obj); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// addRequiredLabelsForPartOfCheObjects searches for secrets and config maps label in Che installation namespace,
+// that have 'app.kubernetes.io/part-of=che' label.
+// When such a resource is found, a standard set of labels is added to the resource to have it cached by the Operator client.
+func addRequiredLabelsForPartOfCheObjects(ctx *deploy.DeployContext) error {
+	cheFlavor := deploy.DefaultCheFlavor(ctx.CheCluster)
+
+	// Prepare selector for all part-of objects in the installation namespace
+	partOfCheSelectorRequirement, err := labels.NewRequirement(deploy.KubernetesPartOfLabelKey, selection.Equals, []string{deploy.CheEclipseOrg})
+	if err != nil {
+		logrus.Error(getFailedToCreateSelectorErrorMessage())
+		return err
+	}
+	instanceNotCheFlavorSelectorRequirement, err := labels.NewRequirement(deploy.KubernetesInstanceLabelKey, selection.NotEquals, []string{cheFlavor})
+	if err != nil {
+		logrus.Error(getFailedToCreateSelectorErrorMessage())
+		return err
+	}
+	objectsToMigrateLabelSelector := labels.NewSelector().Add(*partOfCheSelectorRequirement).Add(*instanceNotCheFlavorSelectorRequirement)
+	listOptions := &client.ListOptions{
+		LabelSelector: objectsToMigrateLabelSelector,
+		Namespace:     ctx.CheCluster.GetNamespace(),
+	}
+
+	// Migrate all config maps with part-of che label
+	configMapsList := &corev1.ConfigMapList{}
+	if err = ctx.ClusterAPI.NonCachingClient.List(context.TODO(), configMapsList, listOptions); err != nil {
+		logrus.Warn(getFailedToGetErrorMessageFor("Config Maps"))
+		return err
+	}
+	if configMapsList.Items != nil {
+		for _, cm := range configMapsList.Items {
+			cm.ObjectMeta.Labels[deploy.KubernetesInstanceLabelKey] = cheFlavor
```

Comment:
Why not to reuse addRequiredLabelsForObject ?

Reply:
`addRequiredLabelsForObject` is intended to get, modify and update the object, where as `addRequiredLabelsForPartOfCheObjects` uses selector to get the whole list and update the items.